### PR TITLE
Handle LogPoller edge case involving backfill followed by an error

### PR DESCRIPTION
**Pull request by reductionista**
When there is a backfill on a brand new chain (no blocks yet in the db) followed by an error, logs get written to the db but no blocks. This will make the logpoller (or backup log poller) rely on the chain next time instead of the db to determine lastFinalizedBlock, which could result in a gap in logs processed. Fixing this by writing the last block in the backfil to the db along with the logs. Its lastFinalizedBlock field will be set to its own block number + 1, so if the db crashes it should start by pulling the logs from that one.

### CODE REVIEW

File: core/chains/evm/logpoller/log_poller_test.go [hunk 1]
```diff
@@ -1057,12 +1058,85 @@ func TestLogPoller_PollAndSaveLogsDeepReorg(t *testing.T) {
...
 		})
 	}
 }
 
+// TestLogPoller_ReorgDuringStartup
+//
+// Tests an edge case where a re-org happens when the number of blocks in the db so far
+// is less than the re-org depth:
+//
+// Orig:  1 <- 2 <- 3 <- 4 <- 5 <- (6)     ( lp starts on block 6 )
```
Comment (connorwstein) [inline core/chains/evm/logpoller/log_poller_test.go:1072]: lp should never start on an unfinalized block though?

File: core/chains/evm/logpoller/log_poller.go [hunk 2]
```diff
@@ -679,7 +679,7 @@ func (lp *logPoller) backfill(ctx context.Context, start, end int64) error {
 		}
 
 		lp.lggr.Debugw("Backfill found logs", "from", from, "to", to, "logs", len(gethLogs), "blocks", blocks)
-		err = lp.orm.InsertLogs(convertLogs(gethLogs, blocks, lp.lggr, lp.ec.ConfiguredChainID()), pg.WithParentCtx(ctx))
+		err = lp.orm.InsertLogsWithBlock(convertLogs(gethLogs, blocks, lp.lggr, lp.ec.ConfiguredChainID()), blocks[len(blocks)-1], pg.WithParentCtx(ctx))
```
Comment (connorwstein) [inline core/chains/evm/logpoller/log_poller.go:682]: 👍

**Review by connorwstein — APPROVED** [review]
